### ISSUE TITLE
Support Mathematica 9 MathLink directory location on Mac OS Create issue

### ISSUE THREAD

**Issue by barrywardell** [SimulationTools/h5mma#2]
Mathematica 9 requires
```Makefile
MLINKDIR ?= /Applications/Mathematica.app/SystemFiles/Links/MathLink/DeveloperKit/MacOSX-x86-64/CompilerAdditions
```
instead of
```Makefile
MLINKDIR ?= /Applications/Mathematica.app/SystemFiles/Links/MathLink/DeveloperKit/CompilerAdditions
```
in the Makefile. We should use whichever of these exists.

> Imported from: [BitBucket Issue #2](https://bitbucket.org/simulationtools/h5mma/issues/2/support-mathematica-9-mathlink-directory)
> Original date: 2013-08-17
> Original creator: Barry Wardell

**Comment by barrywardell** [comment]
This sort of logic is difficult to achieve in a Makefile. Should we be thinking about using autoconf or similar? That also has the advantage that it is more familiar to people than editing the makefile fragment.

> Original date: 2013-08-19
> Original author: Ian Hinder

**Comment by barrywardell** [comment]
This was fixed in 20df41bbc2f0f142452ce6f4365ff9edc0550073. The fix was to use the path required by Mathematica 9 and 10, so compiling with earlier versions requires MLINKDIR to be set in make.defs.

> Original date: 2015-02-26
> Original author: Barry Wardell
